napari/utils/events/evented_model.py

```diff
@@ -91,6 +97,26 @@ def __new__(mcs, name, bases, namespace, **kwargs):
                 # required for pydantic>=1.8.0 due to:
                 # https://github.com/samuelcolvin/pydantic/pull/2064
                 EventedModel.__config__.json_encoders[f.type_] = encoder
+        # check for @_.setters defined on the class, so we can allow them
+        # in EventedModel.__setattr__
+        cls.__property_setters__ = {}
+        for name, attr in namespace.items():
```

Comment:
What happens when a field and a property (with a setter) share the same name? My guess is that the Python property overwrites the Pydantic field, which seems to be consistent with what happens when the property just has a getter (see below), so I think it would be fine.

```python
In [22]: class MyModel(BaseModel):
    ...:     a: int = 1
    ...:     @property
    ...:     def a(self):
    ...:         return 2
    ...: 

In [23]: m = MyModel()

In [24]: m.a
Out[24]: 2
```

Just asking to check and to see if you think it's worth raising an exception here in that case.

Reply:
> My guess is that the Python property overwrites the Pydantic field, which seems to be consistent with what happens when the property just has a getter (see below)

yep, correct.  You're basically changing the namespace before the type gets created

Reply:
one way we could catch that is to look in `MyModel.__annotations__`... and make sure that the name `a` isn't in annotations... but it seems like a pretty rare case

Reply:
Yep, I think it's fine to ignore for now.